catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/GeoPdfParser.java

```diff
@@ -0,0 +1,298 @@
+            return wkt.toString();
+        } else {
+            return polygons.stream()
+                    .map(polygon -> "((" + polygon + "))")
+                    .collect(Collectors.joining(",", MULTIPOLYGON, ")"));
+        }
+    }
+
+    /**
+     * A PDF Neatline defines the area of a PDF image with relation to the PDF page.  If no neatline is given
+     * it is assumed that the image encompasses the entire page.  This functiong generates a NeatLine
+     * in this fashion.
+     * @param pdPage the page to generate the NeatLine
+     * @return an array of points representing the NeatLine
+     */
+    private COSArray generateNeatLineFromPDFDimensions(PDPage pdPage) {
+        COSArray neatLineArray = new COSArray();
+
+        String width = pdPage.getMediaBox().getWidth() + "";
+        String height = pdPage.getMediaBox().getHeight() + "";
+
+        neatLineArray.add(new COSString("0"));
+        neatLineArray.add(new COSString("0"));
+
+        neatLineArray.add(new COSString(width));
+        neatLineArray.add(new COSString("0"));
+
+        neatLineArray.add(new COSString(width));
+        neatLineArray.add(new COSString(height));
+
+        neatLineArray.add(new COSString("0"));
+        neatLineArray.add(new COSString(height));
+
+        return neatLineArray;
+    }
+
+    /**
+     * Convert a Point2d into WKT Lat/Lon
+     *
+     * @param point2D
+     * @return a String representation of a WKT Lat/Lon pair
+     */
+    private String point2dToWkt(Point2D point2D) {
+        return point2D.getX() + " " + point2D.getY();
+    }
+
+    /**
+     * Parses a given NeatLine and Transformation matrix into a WKT String
+     *
+     * @param lgidict         - The PDF's LGIDict object
+     * @param neatLineArray   - The NeatLine array of points for the PDF
+     * @param toDoubleVisitor - A visitor that converts PDF Strings / Ints / Longs into doubles.
+     * @return the generated WKT Lat/Lon set
+     * @throws IOException
+     */
+    private String getWktFromNeatLine(COSDictionary lgidict, COSArray neatLineArray,
+            ICOSVisitor toDoubleVisitor) throws IOException {
+        List<Double> neatline = new LinkedList<>();
+        StringBuilder wktString = new StringBuilder();
+        List<String> coordinateList = new LinkedList<>();
+        String firstCoordinate = null;
+
+        double[] points = new double[NEATLINE_SIZE];
+        for (int i = 0; i < 6; i++) {
+            points[i] = (Double) lgidict.getObjectFromPath(CTM + "/[" + i + "]")
+                    .accept(toDoubleVisitor);
+        }
+        AffineTransform affineTransform = new AffineTransform(points);
+
+        for (int i = 0; i < neatLineArray.size(); i++) {
+            neatline.add((Double) neatLineArray.get(i)
+                    .accept(toDoubleVisitor));
+        }
+
+        for (int i = 0; i < neatline.size(); i += 2) {
+            double x = neatline.get(i);
+            double y = neatline.get(i + 1);
```

Comment:
neatline is an array of points in the form (x,y,x,y) and should never have an odd number of elements